src/sparse_merkle_tree/smt.rs

```diff
@@ -863,6 +863,9 @@ impl SparseMerkleTree {
             if utils::is_bytes_equal(key, query.key()) {
                 continue;
             }
+            if query.bitmap.len() > 0 && query.bitmap[0] == 0 {
```

Comment:
We do not need to check `zero` value at the end of the `bitmap`, because the issue mentioned at the begging or the end.

Reply:
it was actually valid to have ending `0` with the bitmap, but it should fail if we append `0` (which becomes invalid)